src/test/java/tests/MilestonePickerStateTests.java

```diff
@@ -9,46 +9,57 @@
     public MilestonePickerState prepareUnassignedState() {
-        List<PickerMilestone> labelList = new ArrayList<>();
-        PickerMilestone label = new PickerMilestone(new TurboMilestone("testrepo", 1, "milestone1"));
-        labelList.add(label);
-        return new MilestonePickerState(labelList);
+        List<PickerMilestone> milestoneList = new ArrayList<>();
+        PickerMilestone milestone = new PickerMilestone(new TurboMilestone("testrepo", 1, "milestone1"));
+        milestoneList.add(milestone);
+        return new MilestonePickerState(milestoneList, "");
     }
 
     public MilestonePickerState prepareAssignedState() {
-        List<PickerMilestone> labelList = new ArrayList<>();
-        PickerMilestone label = new PickerMilestone(new TurboMilestone("testrepo", 1, "milestone1"));
-        PickerMilestone label2 = new PickerMilestone(new TurboMilestone("testrepo", 1, "milestone2"));
-        label2.setSelected(true);
-        labelList.add(label);
-        labelList.add(label2);
-        return new MilestonePickerState(labelList);
+        List<PickerMilestone> milestoneList = new ArrayList<>();
+        PickerMilestone milestone1 = new PickerMilestone(new TurboMilestone("testrepo", 1, "milestone1"));
+        PickerMilestone milestone2 = new PickerMilestone(new TurboMilestone("testrepo", 2, "milestone2"));
+        milestoneList.add(milestone1);
+        milestoneList.add(milestone2);
+        // userInput is set to 'milestone2'
+        return new MilestonePickerState(milestoneList, milestone2.getTitle());
     }
 
     @Test
     public void toggleMilestone_noMilestone_milestoneAssigned() {
         MilestonePickerState state = prepareUnassignedState();
         state.toggleExactMatchMilestone("milestone1");
-        assertTrue(state.getCurrentMilestonesList().get(0).isSelected());
+        assertTrue(state.getAllMilestones().get(0).isSelected());
     }
 
     @Test
     public void toggleMilestone_hasMilestone_milestoneReplaced() {
         MilestonePickerState state = prepareAssignedState();
         state.toggleExactMatchMilestone("milestone1");
-        assertTrue(state.getCurrentMilestonesList().get(0).isSelected());
-        assertFalse(state.getCurrentMilestonesList().get(1).isSelected());
+        assertTrue(state.getAllMilestones().get(0).isSelected());
+        assertFalse(state.getAllMilestones().get(1).isSelected());
     }
 
     @Test
     public void toggleMilestone_hasMilestone_milestoneUnassigned() {
         MilestonePickerState state = prepareAssignedState();
         state.toggleExactMatchMilestone("milestone2");
-        assertFalse(state.getCurrentMilestonesList().get(0).isSelected());
-        assertFalse(state.getCurrentMilestonesList().get(1).isSelected());
+        assertFalse(state.getAllMilestones().get(0).isSelected());
+        assertFalse(state.getAllMilestones().get(1).isSelected());
+    }
+
+    @Test
+    public void bestMatches_withUserInput_previousMatchingMilestoneAppears() {
```

Comment:
`previousMatchingMilestoneIncluded` sounds more accurate. `Appears` suggests this is some UI state test, which isn't.